### ISSUE TITLE
[Retraining] Use Liger Kernel to avoid multi-head logits materialization and scale the context length by N times

### ISSUE THREAD

**Issue by ByronHsu** [FasterDecoding/Medusa#119]
https://github.com/linkedin/Liger-Kernel/tree/main/examples/medusa

With the implementation of FusedLinearCrossEntropy and other kernels in Liger-Kernel, we are able to effectively reduce the memory while increase the throughput. We are happy to collaborate and integrate with our kernels! 

![image](https://github.com/user-attachments/assets/da70dff7-e0b7-4be1-8b8c-a83a4ab82f8f)
![image](https://github.com/user-attachments/assets/3719d649-1210-43e3-919a-cd1f3850ea25)

**Comment by ByronHsu** [comment]
cc @ctlllll @leeyeehoo @zhyncs
